### ISSUE TITLE
statistics gathering and reporting

### ISSUE THREAD

**Issue by MrDys** [projectblacklight/blacklight#195]
CODEBASE-164: Would be great to have usage data for searches, etc.   Could do for jetty solr for now ?   Buckley Amendment fun???

**Comment by mejackreed** [comment]
With the current use of Google Analytics by many, is this a feature we need to add to Blacklight? Still relevant?

**Comment by cbeer** [comment]
I think there's additional data people wish we could collect (@jkeck, @jronallo ?), like:

- which facets were applied
- what rank were the documents users clicked on

IIRC, some of this can be done by adding better `id`/`class`/`data` attributes to elements to make it easier to trigger custom events from javascript-land.

**Comment by jkeck** [comment]
Yeah, I don't know enough about how to do custom events in GA these days, but we could probably do something better about giving people the hooks to filter based on parts of the UI (so that those who would like to gather these types of stats don't have to do heavy UI customization).

**Comment by jronallo** [comment]
This code is old but might still be useful for someone who wants to
implement GA event tracking in Blacklight. I'd be willing to let folks push
to the repo (or move it to under project blacklight) and add a owners on
rubygems.org.

https://github.com/jronallo/blacklight_google_analytics

Jason
